src/main/java/com/example/videotecha/model/RegisteredUser.java

```diff
@@ -0,0 +1,127 @@
+package com.example.videotecha.model;
+
+import com.example.videotecha.model.enums.UserRole;
+import jakarta.persistence.*;
+
+import java.util.Objects;
+
+@Entity
+public class RegisteredUser {
+
+    @Id
+    @SequenceGenerator(name = "userSequenceGenerator", sequenceName = "userSequenceGenerator", initialValue = 1, allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "userSequenceGenerator")
+    @Column(name="reg_user_id", unique=true, nullable=false)
+    private Long id;
+
+    @Column(nullable=false)
+    private String firstName;
+
+    @Column(nullable=false)
+    private String lastName;
+
+    @Column(unique=true, nullable=false)
+    private String email;
+
+    @Column(unique=true, nullable=false)
+    private String username;
+
+    @Column(nullable=false)
+    private String password;
+
+    @Column(nullable=false)
+    private UserRole role;
+
+    public RegisteredUser() {}
+
+    public RegisteredUser(String firstName, String lastName, String email, String username, String password, UserRole role) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.email = email;
+        this.username = username;
+        this.password = password;
+        this.role = role;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public UserRole getRole() {
+        return role;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public void setRole(UserRole role) {
+        this.role = role;
```

Comment:
If user roles are not to be changed, setters are redundant. Same goes for other setters.